### PR TITLE
Inform player if they are capable of learning a recipe from disassembly.

### PR DESCRIPTION
**Pull request by DoctorGoat**
DracoGriffin from the Discord wanted this feature, so here it is. It's fairly simple; it'd probably also be an approach to append another string based on skill level/learnability/other factors of learning a recipe from disassembly.

edit: i am not used to astyle. or github

### CODE REVIEW

**Comment by DracoGriffin** [comment]
Found it useful for my illiterate playthrough so I didn't have to constantly reference jsons if items were "reversible: true".

**Comment by DoctorGoat** [comment]
Could someone more expeerienced with it than me help with the astyle check? I ran crafting.cpp through astyle specifically after the first time it failed and it hasn't improved anything.

File: src/crafting.cpp [hunk 1]
```diff
@@ -1113,8 +1113,15 @@ bool player::disassemble( item &obj, int pos, bool ground, bool interactive )
             list << "- " << elem.front().to_string() << std::endl;
         }
 
-        if( !query_yn( _( "Disassembling the %s may yield:\n%s\nReally disassemble?" ), obj.tname().c_str(),
-                       list.str().c_str() ) ) {
+        if ( !r.learn_by_disassembly.empty() && !knows_recipe( &r ) && can_decomp_learn( r ) )
```
Comment (Night-Pryanik) [inline src/crafting.cpp]: Should be like `if( foo ) {` : `if` is glued to the opening bracket; curly bracket is on the same line as `if`.

File: src/crafting.cpp [hunk 2]
```diff
@@ -1113,8 +1113,15 @@ bool player::disassemble( item &obj, int pos, bool ground, bool interactive )
...
+        {
+            if ( !query_yn(_("Disassembling the %s may yield:\n%s\nReally disassemble?\nYou feel you may be able to understand this object's construction.\n"), obj.tname().c_str(),
+                           list.str().c_str() ) ) {
+                return false;
+            }
+        }
+        else if ( !query_yn(_("Disassembling the %s may yield:\n%s\nReally disassemble?"), obj.tname().c_str(),
```
Comment (Night-Pryanik) [inline src/crafting.cpp]: Should be like `} else if( foo`.

File: src/crafting.cpp [hunk 3]
```diff
@@ -1113,8 +1113,13 @@ bool player::disassemble( item &obj, int pos, bool ground, bool interactive )
             list << "- " << elem.front().to_string() << std::endl;
         }
 
-        if( !query_yn( _( "Disassembling the %s may yield:\n%s\nReally disassemble?" ), obj.tname().c_str(),
-                       list.str().c_str() ) ) {
+        if( !r.learn_by_disassembly.empty() && !knows_recipe( &r ) && can_decomp_learn( r ) ) {
+            if( !query_yn(_("Disassembling the %s may yield:\n%s\nReally disassemble?\nYou feel you may be able to understand this object's construction.\n"), obj.tname().c_str(),
```
Comment (ZhilkinSerg) [inline src/crafting.cpp]: Need astyle.

**Pull request merged**